escher/src/text_area.cpp

```diff
@@ -307,8 +313,12 @@ TextArea::TextArea(Responder * parentResponder, char * textBuffer,
 bool TextArea::TextArea::handleEvent(Ion::Events::Event event) {
   if (event == Ion::Events::Left) {
     m_contentView.moveCursorIndex(-1);
+  } else if (event == Ion::Events::Origin) {
+    m_contentView.moveCursorIndexAbsolute(0);
   } else if (event == Ion::Events::Right) {
     m_contentView.moveCursorIndex(1);
+  } else if (event == Ion::Events::End) {
+    m_contentView.moveCursorIndexAbsolute(INT_MAX);
```

Comment:
m_contentView.moveCursorGeo(INT_MAX, 0);

Reply:
If you do that, you will have to correct the following method:
size_t TextArea::Text::indexAtPosition(Position p) {
  if (p.line() < 0) {
    return 0;
  }
  int y = 0;
  const char * endOfLastLine = nullptr;
  for (Line l : *this) {
    if (p.line() == y) {
      size_t x = max(p.column(),  0);
      x = min(clippedPColumn, l.length());
      return l.text() - m_buffer + x;
    }
    endOfLastLine = l.text() + l.length();
    y++;
  }
  assert(endOfLastLine != nullptr && endOfLastLine >= m_buffer);
  return endOfLastLine - m_buffer;
}